### ISSUE TITLE
Finish LSETest.m

### ISSUE THREAD

**Issue by jtorre39** [kdahlquist/GRNmap#376]
An incomplete test function for LSE was found. The solution is for the coding team to pick up where the previous authors left off and finish it up. The test file can be found in GRNmap/test_files/calculationTests/LSETest.m

**Comment by jtorre39** [comment]
Found issue #313 which was closed regarding creating tests for LSE.

**Comment by jtorre39** [comment]
According to #313 we seem to have left LSETest.m incomplete as it was superseded by the completion of GeneralLSETest.m

**Comment by jtorre39** [comment]
![image](https://user-images.githubusercontent.com/16827894/31975553-cae80c30-b8e6-11e7-950a-f8b1494cd3b9.png)

This is the most recent code coverage report as of 10/24/2017

**Comment by jtorre39** [comment]
I had some time and ran the full test suite between classes. The code coverage for lse.m went up a trivial amount. But the point was to indeed confirm that lse.m gets tested and to ensure coverage for it did not go down as we made more tests.

**Comment by jtorre39** [comment]
#313 We made this list of variables that are candidates for testing within lse.m.

- initial_guesses
- estimated_guesses
- lse_0
- lse_final - which is accounted for in General Least Squares Error and does not need to be tested independently

Since we've completed tests for General Least Squares Error, lse_final would not need to be tested. However, it is suggested through this comment that the rest of the bullet list is not tested for anywhere else and should be accounted for within a unit test.

**Comment by kdahlquist** [comment]
@jtorre39 walked us through the lse.m code to indicate where `estimated_guesses` may be difficult to test in the case where `estimate_params` is true, because in that situation `fmincon` is run and hand-calculated or analytic solutions may not be feasible.

For now, @jtorre39 should just write tests for the case where `estimate_params` is false because that is when `initial_guesses` should be equal to `estimated_guesses`. @bengfitzpatrick will mull over the question of what to do when `estimate_params` is true.

**Comment by jtorre39** [comment]
Finished all but 1 test cases for LSETest.m. Test cases have not required much code do to the tests we wrote validating GLSE's functionality as LSE for the most part simply calls GLSE in various ways. Initial_Guesses is calculated within LSE however, so I will have to take some time understanding the code so as to hand calculate a value.

**Comment by jtorre39** [comment]
Another task might be to create a test struct within the test file for LSE as it currently runs an input sheet through readinputsheet which takes some time and will add up as we continue to run the test suite in the future.

**Comment by dondi** [comment]
This is nearly done, with the following notes:
* Many tests are pass-throughs to general LSE, so that test suite will be relied on to cover those.
* @jtorre39 has one more test that does need to be written at this layer, having to do with `initialGuesses`. That is anticipated to be completed by next week or the week after.

**Comment by jtorre39** [comment]
Finished LSETest. The changes can be viewed on my fork. I'll see if I get some extra time before the next meeting to eliminate the need of an input sheet within the test to speed up testing a bit. If I can't in time I'll write up a new issue for it.

**Comment by jtorre39** [comment]
I spent some time wrangling down a failing test after removing the use of an input file and creating a constant GRNstruct in the test file itself. It came down to a simple precision error. I updated the test accordingly and LSETest can be considered finished for now.

**Comment by jtorre39** [comment]
I did however seem to find another bug related to compressmissingdata in my tracking down of my precision bug. I will post this in issue #359 for discussion.

**Comment by dondi** [comment]
Thanks for the update (from a few thousand feet in the air)! I looked at the pull request and it looks good to me. Good job eliminating the need for the input file. The tests look good also.

If Dr. Fitzpatrick is at the meeting, you can show him the relevant parts of the PR to see if this works for him too. Otherwise I am good with closing this issue. (I agree that the new compressmissingdata problem is an issue of its own—see comments in #359)

**Comment by kdahlquist** [comment]
This is closeable upon PR #378 being accepted.

**Comment by dondi** [comment]
If @bengfitzpatrick is OK with the code (or if he was not present at the meeting), I can go ahead and merge the PR.

**Comment by jtorre39** [comment]
Removing lse_tests folder as the input sheet is no longer used. Also, @dondi we're waiting on merging my PR because it also includes work on the unfinished test file audit issue #361.

**Comment by dondi** [comment]
PR has been merged; closing this now.
